plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplit.java

```diff
@@ -255,6 +260,25 @@ public SplitWeight getSplitWeight()
         return splitWeight;
     }
 
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE
+                + estimatedSizeOf(path)
+                + estimatedSizeOf(schema, key -> estimatedSizeOf((String) key), value -> estimatedSizeOf((String) value))
```

Comment:
there is some more stuff in `Properties` (`defaults`, `map`) - not sure if set in our case.

Reply:
This method should account memory for all key-value pairs stored in the `Properties`. However as you pointed out it may not account for all internal data structure maintained by the `map` (what duplicates the `Properties` map for efficiency) and `defaults` (which may shadow some of the keys). It is hard to estimate memory usage of `Properties` precisely, but the current estimate should be close. Also the `schema` is likely to be shared between many splits, so we are likely over-counting anyway. 

Ideally the `schema` field should be removed completely, as it duplicates a lot of data already present in the split (https://github.com/prestodb/presto/pull/13453). But that's probably a topic for a follow up.